language/move-lang/tests/move_check/typing/module_call_constraints_not_satisfied.exp

```diff
@@ -1,282 +1,312 @@
+33 │         both(new_box<C>(), new_box<R>())
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   │         │                          │
+   │         │                          The type '0x8675309::M::Box<R>' can have the ability 'copy' but the type argument 'R' does not have the required ability 'copy'
    │         'copy' constraint not satisifed
 
 error[E05001]: ability constraint not satisfied
-   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:30:9
+   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:37:9
    │
- 5 │     struct Box3<T1, T2, T3> has copy, drop {}
+ 5 │     struct Box3<T1, T2, T3> has copy, drop { f1: T1, f2: T2, f3: T3 }
    │            ---- To satisfy the constraint, the 'key' ability would need to be added here
    ·
-15 │     fun rsrc<R: key>(r: R) {
+11 │     fun new_box3<T1, T2, T3>(): Box3<T1, T2, T3> {
+   │                                 ---------------- The type '0x8675309::M::Box3<C, C, C>' does not have the ability 'key'
+   ·
+23 │     fun rsrc<R: key>(r: R) {
    │                 --- 'key' constraint declared here
    ·
-30 │         rsrc(Box3<C, C, C> {});
-   │         ^^^^^^^^^^^^^^^^^^^^^^
-   │         │    │
-   │         │    The type '0x8675309::M::Box3<C, C, C>' does not have the ability 'key'
-   │         'key' constraint not satisifed
+37 │         rsrc(new_box3<C, C, C>());
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^ 'key' constraint not satisifed
 
 error[E05001]: ability constraint not satisfied
-   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:32:9
+   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:39:9
    │
-11 │     fun cpy<C: copy>(c: C) {
+11 │     fun new_box3<T1, T2, T3>(): Box3<T1, T2, T3> {
+   │                                 ---------------- The type '0x8675309::M::Box3<R, C, C>' does not have the ability 'copy'
+   ·
+19 │     fun cpy<C: copy>(c: C) {
    │                ---- 'copy' constraint declared here
    ·
-32 │         cpy(Box3<R, C, C> {});
-   │         ^^^^^^^^^^^^^^^^^^^^^
-   │         │   │    │
-   │         │   │    The type '0x8675309::M::Box3<R, C, C>' can have the ability 'copy' but the type argument 'R' does not have the required ability 'copy'
-   │         │   The type '0x8675309::M::Box3<R, C, C>' does not have the ability 'copy'
+39 │         cpy(new_box3<R, C, C>());
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^
+   │         │            │
+   │         │            The type '0x8675309::M::Box3<R, C, C>' can have the ability 'copy' but the type argument 'R' does not have the required ability 'copy'
    │         'copy' constraint not satisifed
 
 error[E05001]: ability constraint not satisfied
-   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:33:9
+   ┌─ tests/move_check/typing/module_call_constraints_not_satisfied.move:40:9
    │
-11 │     fun cpy<C: copy>(c: C) {
+11 │     fun new_box3<T1, T2, T3>(): Box3<T1, T2, T3> {
```

Comment:
A change for another PR but this error message should really be improved. Repeats itself a bit and could point to the type `'C'` or whatever type doesn't have the ability